duckduckgo/iOS
# Make voice search opt in

Task/Issue URL: https://app.asana.com/0/414235014887631/1202533216912528/f

**Description**:
We're changing the default behavior of voice-search. Before it was always on, now it's an opt-in feature where users can turn it on/off via Settings.

**Steps to test this PR**:
1. After fresh install, confirm the voice-search icon is not visible
2. Go to settings, enable "Private Voice Search", accept the mic permission
3. Close settings and confirm the voice-search icon is visible and working

------
1. After fresh install, confirm the voice-search icon is not visible
2. Go to settings, enable "Private Voice Search", reject the permission
3. Close settings and confirm the voice-search icon is not visible
4. Go back to settings, try to enable "Private Voice Search" again, confirm the alert with the settings CTA is working
5. Enable mic permissions and enable the feature
6. Close settings and confirm the voice-search icon is visible and working

------
Confirm that the migration is working. The way the migration works is:
1. If the user had previously accepted microphone permission, the voice-search should be toggled on as default
2. if the user had previously rejected or never granted microphone permission, the voice-search should be toggled off as default

-----
Make sure there's no mention of voice-search on settings on iOS 14 since this is a iOS 15+ feature only

**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
